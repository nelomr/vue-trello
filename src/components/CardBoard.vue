<template>
    <li class="card-board">
        <div class="card-board--title">
            <input
                @input="changeTitle()"
                type="text"
                v-model="title"
            >
        </div>
        <div class="card-board--content">
            <div class="card-board--description">
                <textarea
                    @input="changeDescription()"
                    resize="false"
                    v-model="description"
                    placeholder="Add some description..."
                    type="text"
                />
                <div class="card-board--buttons">
                    <a href="#" @click.prevent="'#'"><font-awesome-icon icon="trash-alt" size="1x" /></a>
                </div>
            </div>
        </div>
    </li>
</template>

<script>
import { mapGetters } from 'vuex';
export default {
    components: {
    },
    props: {
        task: {
            type: Object,
            default: () => ({})
        }
    },
    data() {
        return {
            enableEdit: false,
            title: this.task.name,
            description: this.task.description
        }
    },
    computed: {
        ...mapGetters(['getTask']),
        getCurrentTask() {
            return this.getTask(this.task.id);
        }
    },
    methods: {
        addDescription() {
            this.enableEdit = !this.enableEdit;
        },
        changeDescription() {
            this.$store.dispatch('setDescriptionTask', {description: this.description, id: this.task.id});
        },
        changeTitle() {
            this.$store.dispatch('setTitleTask', {title: this.title, id: this.task.id});
        }
    }
}
</script>

<style lang="scss">
    .card-board {
        padding: 1em;
        margin-bottom: 30px;
        background-color: $light-color;
        list-style: none;
        border-color: $light-color;
        border-radius: .5em;
        box-shadow: 0 1px 3px 0 rgba(0,0,0,.1), 0 1px 2px 0 rgba(0,0,0,.06);

        &:last-child {
            margin-bottom: 0;
        }
    }

    .card-board--title input {
        font-size: 16px;
        font-weight: bold;
    }

    .card-board--buttons {
        display: flex;
        justify-content: flex-end;

        a {
            display: block;
            color: $secondary-color;
        }
    }
</style>