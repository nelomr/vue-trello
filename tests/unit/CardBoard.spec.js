import Vue from 'vue';
import { shallowMount } from '@vue/test-utils';
import CardBoard from '../../src/components/CardBoard';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'

Vue.component('FontAwesomeIcon', FontAwesomeIcon);

describe('CardBoard', () => {
    const build = () => {
        const wrapper = shallowMount(CardBoard);

        return {
            wrapper
        };
    };
    
    it('renders the component', () => {
        const {wrapper} = build();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
