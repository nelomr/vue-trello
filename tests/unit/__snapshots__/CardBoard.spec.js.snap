// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBoard renders the component 1`] = `
<li class="card-board">
  <div class="card-board--title"><input type="text" maxlength="40"></div>
  <div class="card-board--content">
    <div class="card-board--description"><textarea resize="false" placeholder="Add some description..." type="text"></textarea>
      <div class="card-board--buttons"><a href="#">
          <font-awesome-icon-stub icon="trash-alt" size="1x"></font-awesome-icon-stub>
        </a></div>
    </div>
  </div>
  <popupelement-stub></popupelement-stub>
</li>
`;
